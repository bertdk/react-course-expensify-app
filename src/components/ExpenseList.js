import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
          props.expenses.map((i) => {
            return <ExpenseListItem key={i.id} {...i} />
          })
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filter)
  }
};

export default connect((state) => mapStateToProps)(ExpenseList);