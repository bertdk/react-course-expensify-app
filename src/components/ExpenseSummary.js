import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses';
import getTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({count, total}) => (
  <div>
    <p>Total is {numeral(total / 100).format('$0,0.00')} for {count} expenses</p>
  </div>
);

const mapStateToProps = (state) => {
  return {
    count: selectExpenses(state.expenses, state.filter).length,
    total: getTotal(selectExpenses(state.expenses, state.filter))
  }
};

export default connect((state) => mapStateToProps)(ExpenseSummary);