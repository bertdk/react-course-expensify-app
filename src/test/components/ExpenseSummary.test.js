import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Viewing 1 expenses', () => {
  const wrapper = shallow(<ExpenseSummary count={1} total={333333} />);
  expect(wrapper).toMatchSnapshot();
});

test('Viewing 2 expenses', () => {
  const wrapper = shallow(<ExpenseSummary count={4} total={777777} />);
  expect(wrapper).toMatchSnapshot();
});