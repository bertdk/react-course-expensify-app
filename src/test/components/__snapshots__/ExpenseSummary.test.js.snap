// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewing 1 expenses 1`] = `
<div>
  <p>
    Total is 
    $3,333.33
     for 
    1
     expenses
  </p>
</div>
`;

exports[`Viewing 2 expenses 1`] = `
<div>
  <p>
    Total is 
    $7,777.77
     for 
    4
     expenses
  </p>
</div>
`;
